crates/ruma-state-res/src/lib.rs

```diff
@@ -462,14 +462,15 @@ fn iterative_auth_check<E: Event + Clone>(
         let mut auth_events = StateMap::new();
         for aid in event.auth_events() {
             if let Some(ev) = fetch_event(aid.borrow()) {
-                // TODO synapse check "rejected_reason" which is most likely
-                // related to soft-failing
-                auth_events.insert(
-                    ev.event_type().with_state_key(ev.state_key().ok_or(Error::MissingStateKey)?),
-                    ev,
-                );
+                if !ev.rejected() {
```

Comment:
Assuming the user filters incoming events through `check_state_independent_auth_rules` first, we should never have `ev.rejected()`, right? I'm not sure I understand the purpose of checking this here in that case, although it does seem to match the spec text. Perhaps this is something we should ask the spec people about.

Reply:
I agree with you, the event should have already passed all the checks on receipt of a PDU in theory so it shouldn't be possible to get a rejected event here.

This is the same behavior as Synapse though: https://github.com/element-hq/synapse/blob/740fc885cd593187cfefa599927ce47a5556ffb0/synapse/state/v2.py#L570